### DISCUSSION TITLE
app_commands options

### DISCUSSION THREAD

**Discussion by jshaw-creovai** [Rapptz/discord.py#9495] [Q&A] [answered]
Hi all, I'm currently in process of writing a app command which mostly works fine, but feel like im really struggling/misreading the documentation for options.

So anyway what I am trying to achieve is an lfg bot just to keep my mind busy and what i am failing is making it so whena user runs the command they can add in their own text string so for example: /tlfg (options box shows up named title that can be type in) "my jail raid" but yeah, failing miserably tbh.

```
    @app_commands.command(
        name="tlfg", 
        description="Create a new raid",
        # options=[
        #     discord.app_commands.Option(
        #         name="title",
        #         description="Enter your title here",
        #         type=3,
        #         required=True
        #     ),
        # ]
        )
```
I was wondering someone could advise or give me pointers on what i'm doing wrong.  At this bit "discord.app_commands.Option" I have tried it as jsut app_commands.Option and even just Option to no avail either.

Cheers

**Comment by LostLuma** [comment, marked as answer]
App commands in discord.py specify their parameters via Python parameters rather than having to manually specify them in the decorator. The different types of options can be used by placing their respective type into the type hint for the parameter. See two short examples here:

```py
# Turns into a command called /example_one with a required string as an option
@client.tree.command()
async def example_one(interaction: discord.Interaction, example: str) -> None:
    ...

# Turns into a command called /example_two which requires choosing a server member to use
@client.tree.command()
async def example_two(interaction: discord.Interaction, example: discord.Member) -> None:
    ...
```

See more examples here: https://github.com/Rapptz/discord.py/blob/master/examples/app_commands
Also note the usage of additional decorators like the [describe](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.describe) one to give options descriptions or rename them etc.

**Reply by jshaw-creovai** [reply]
Oh thank you very much, Jumped into the describe optiion and that is doing just what I need, I'll take another look over docs again as well for future references, thank you once again.
